### ISSUE TITLE
Forecast Model Builder - roadmap: more model algorithm options

### ISSUE THREAD

**Issue by sfc-gh-kmason** [Snowflake-Labs/emerging-solutions-toolbox#250]
It would be nice for users to be able to select models other than XGBoost. It might also be nice to be able to select more than one and be able to compare the models against each other. 

Some common time series model choices: scikit-learn, xgboost, LightGBM, Prophet, and ARIMA.
It might be easiest to start with the first three, since Prophet and ARIMA are slightly different to setup and infer with. 

For model comparisons, the user could have the option at the top of the notebook to select a single model or multiple models to try. If the user selects multiple models the code could automatically adjust within the training UDTF to set aside a test set and evaluate the different models against each other.

Could potentially use @sfc-gh-twhite 's work with [mlforecast](https://nixtlaverse.nixtla.io/mlforecast/index.html).

Linking appropriate docs here for the model types:
- https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html
- https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBRegressor
- https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMRegressor.html
- https://facebook.github.io/prophet/docs/quick_start.html#python-api
- https://www.statsmodels.org/stable/generated/statsmodels.tsa.arima.model.ARIMA.html
